### PR TITLE
Adds .gitattributes file to templates

### LINKED ISSUE #8590: New project templates should include a .gitattributes file to prevent EOL conversions
This is related to #8530. It was right to fix autolinking to support modifying solution files with LF or CRLF line endings, because VS doesn't discriminate against either. However, all of the native template files are set to the "default" EOL for the file type according to VS, and while our templating code maintains the EOL of the file as it appears on disk, it's easy and common enough for the EOLs to be changed when checked in to git.

Adding a .gitattributes file to the windows folder of the template, targeting the VS filetypes we add (cpp, h, cs, props, vcxproj, csproj, sln) could help mitigate cross-plat devs hitting future issues.

### PR DESCRIPTION
**Pull request by TatianaKapos**
## Description
Adds .gitattributes to templates (.gitattrubutes file taken from [RNGallery](https://github.com/microsoft/react-native-gallery/blob/main/.gitattributes))

### Type of Change
- New feature (non-breaking change which adds functionality)

### Why
Resolves #8590

### What
Adds .gitattributes file to _vnext/templates_ folder and registers it in _packages/@react-native-windows/cli/src/generator-windows/index.ts_ 

## Testing
Tested locally using steps described in [Contributing to CI Docs](https://github.com/microsoft/react-native-windows/wiki/Contributing-to-the-CLI#4-run-your-local-react-native-windows-init-to-create-the-windows-project) by creating new app/module using local changes

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/9330)

### CODE REVIEW

File: vnext/template/_gitattributes [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# Spec Codegen uses LF
+*Spec.g.h eol=lf
+
+# Force Visual Studio project files (mostly XML) to CRLF
+#   This helps avoid conflict which occurs when Xmarian/Mac contributors with AutoCRLF=input (or off)
+#   check in these files as CRLFs, and then Windows contributors with AutoCRLF=true end up checking then
+#   files in as LFs  (due to AutoCRLF=true normalizing CRLF->LF on check-in).
+
+*.sln text eol=crlf
+*.csproj text eol=crlf
```
Comment (dannyvv) [inline vnext/template/_gitattributes:20]: nit: I often search for msbuild files via `*.*proj`
Wild suggestion: Should we just use that wildcard, or should we list explicitly?

File: vnext/template/_gitattributes [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+*.vbproj text eol=crlf
+*.fsproj text eol=crlf
+*.pyproj text eol=crlf
+*.dbproj text eol=crlf
+*.vcxproj text eol=crlf
+*.shproj text eol=crlf
+*.projitems text eol=crlf
+*.vcxitems text eol=crlf
+*.props text eol=crlf
```
Comment (dannyvv) [inline vnext/template/_gitattributes:29]: maybe I missed it, but `*.targets` are also common msbuild files.

**Review by jonthysell — APPROVED** [review]

**Comment by acoates-ms** [comment]
This change seems pretty broad in its effects.  I'd imagine that forcing all cpp files to CLRF is not something the average cross platform repo wants to have pushed on its iOS/macOS code.

Is there some reason we are forcing such a policy on our users? From the linked bug it sounds like our tools work with LF or CRLF?

**Comment by TatianaKapos** [comment]
The .gitattributes file is placed in the windows folder, so it should only affect windows code. I assumed normalizing the windows line ending to CLRF would help with users from different platforms contribute to the same GitHub repo without checking in different line ending every time, but I believe @jonthysell has more background on this or maybe there's a better way around this!

but I totally see your point of if react-native core doesn't push such requirements, perhaps we shouldn't either.

**Comment by TatianaKapos** [comment]
@acoates-ms Another point made offline is the file can be removed by the user if they don't want it!

**Comment by NickGerleman** [comment]
I agree that I'm not sure this makes sense to have in our template. Our template is getting large, and prone to fragmentation. I'm also not sure we can make assumptions about our user's preferences.

**Comment by TatianaKapos** [comment]
Hmm I wasn't at the triage meeting when this issue was pushed through, but I'm totally fine closing this PR if that's the majority! Chris mentioned @asklar and @jonthysell may have some background on why the issue got filed/pushed through? Just wanted to check in with them before closing the PR/issue :)

**Comment by chrisglein** [comment]
It's been a while (~4 months) since we looked at the issue in triage, but I don't remember pushback then. Why do we feel different about this than say linting rules? My assumption is that this is a default in the app template but also can be ripped out or modified by the app if they care.

Based on the proposal from @jonthysell it sounds like this was suggested to reduce real pain we'd felt in trying to run apps through the update process. So reducing risk to those apps that leave it in place. Am I reading that right?

@NickGerleman @acoates-ms Care to elaborate on the pushback given the info in this thread?

**Comment by jonthysell** [comment]
Because we create/modify source files in our CLI, we've run into issues with modifying files with different line endings. We don't have any such active issues at the moment - adding an .gitattributes could help prevent CLI bugs from messing up user's files, and/or prevent unecessary conversions back and forth when xplat users work on their files.

I'm fine with not adding this to the template if we don't want to claim any further prescriptiveness of the template files.

**Comment by TatianaKapos** [comment]
Closing due to
- Not wanting to claim any further prescriptiveness of template files since our template files are large and prone to fragmentation.
- React-native core doesn't require or push these kind of requirements.
- Our tooling has been since fixed to work with different line endings.

**Pull request closed without merging**